subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -1352,6 +1352,7 @@ int lwm2m_rd_client_stop(struct lwm2m_ctx *client_ctx,
 
 	client.ctx->event_cb = event_cb;
 	rd_client_message_free();
+	lwm2m_engine_context_close(client_ctx);
```

Comment:
What is this one?
I think this is completely unrelated to this PR

Reply:
@SeppoTakalo thanks for your comment.
After some tests, it appears that if there is a network error or other, timeout callback shall be called on client stop to release send ACKs.
Do you think it shall be done on another PR?

Reply:
Yes, at least create a issue ticket that describes the problem.
I think this is related to issue discussed here: https://github.com/zephyrproject-rtos/zephyr/pull/53476#issuecomment-1400491225

So don't blindly fix, because it might fix only one of the corner cases or might introduce side effects. So at least a separate PR is required, but I would prefer reporting the issue first, so we have a record what is going to be fixed.

Reply:
Has been removed : I will open a new issue